README.md
![alt XOOPS CMS](http://xoops.org/images/logoXoops4GithubRepository.png)
##Wordbook Module for [XOOPS CMS 2.5.x](http://xoops.org)

**Wordbook** for  [XOOPS CMS 2.5.x](http://xoops.org) is a flexible and versatile module that uses XOOPS features to provide the infrastructure to build attractive, usable and powerful glossaries, dictionaries, or encyclopedias..

**XOOPS Module Archive:** Module in this archive are older modules for XOOPS CMS, and therefore some of them might not work with the latest XOOPS and PHP versions. 

However, please fork them and if you have any improvements and fixes, please share them back with us. 

Please visit us on http://xoops.org

**The "next generation" of XOOPS CMS is being crafted on GitHub at: https://github.com/XOOPS**


docs/changelog.txt
2.00 Beta 1             2016-01-02
---------------------------------------
- Converted to XOOPS 2.5.7.2 Admin GUI (Mamba)

1.17 RC 1               2006-11-30
---------------------------------------
- BugFix: global search-function was erroneous.
- BugFix: letter.php: query 'show allentries' had no results.
- Bugfix: admin/submissions.php (had no function at all).
- BugFix: offline terms were included in autolinked terms.
- Bugfix: changing category of entries is now possible in xoops versions other than 2.0.13.2.
- Bugfix: include/cleantags.php
- Bugfix: display HTML-tags
- Bugfix: set $xoopsOption['template_main'] before including root/header.php - for caching.
- renamed adminmenu() to wb_adminmenu() to avoid conflicts with other modules (cleanTags dito).
- Changed: adminmenu
- Changed: added icon search field
- Changed: field 'reference' is automatically included in global search.
- Changed: field 'reference' is automatically included in local search in terms & definitions.
- Add: script to keep blocks settings onUpdate
- Add: pagetitles:  eg: Site - Term - Category - Wordbook
- Add: admin/about.php
- Add: french language files
- Add: plugins for modules 'waiting', 'What's New' and 'RSSFit'.


<B>Wordbook v1.16</B>
---------------------------------------
Fix: (by Alama) Grammatical and stylistic corrections to the English language interface to Wordbook 1.15 in main.php and modinfo.php
Fix: Anonymous users can now see sub menu items if appropriated permissions are set.
Fix: Fixed a bug in admin/entry.php. When an administrator modied an entry, became proprietary of entry and the combo box in which he could choose the author was ignored.
Fix: Missing data in results produced by search.php. Not only was the category missing, but the entryID number of the term in the link is missing, making the first entry in the table come up for all word's links clicked.
Modified: funtion getLinkedUnameFromId() in include/functions.php renamed to wb_getLinkedUnameFromId() to avoid possible duplicated function name error.
Add: submit.php page uses now templates and css formating as the other pages do.
Add: Submit option in main menu is shown only if user/anon-user is allowed to submit new words. (users and anonimous users privileges set in config section)

<B>Wordbook v1.15 Beta 2</B>
---------------------------------------
Fix: Fixed a bug in include/functions.php and block/random_term.php that caused browser cant go back if deletion of an entry is canceler (page del icon had a link with target="_blank" parameter instead "_self" or even nothing)
New: Comments feature Added (modified files: moninfo.php entry.php admin/entry.php admin/category.php)
New: Add an Import script from module Dictionary v0.9x (Import comments are made by moving them from Dictionary module to Wordbook module).

1.00                2004-05-08
---------------------------------------
First official version.


0.99                2004-04-10
---------------------------------------
First public release.



xoops_version.php
<?php
/*
 * You may not change or alter any portion of this comment or credits
 * of supporting developers from this source code or any supporting source code
 * which is considered copyrighted (c) material of the original comment or credit authors.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 */

/**
 * @copyright    The XOOPS Project http://sourceforge.net/projects/xoops/
 * @license      GNU GPL 2 or later (http://www.gnu.org/licenses/gpl-2.0.html)
 * @package      Wordbook - a multicategory glossary
 * @since        8 May 2004
 * @author       hsalazar, XOOPS Development Team
 * Additions:    Dario Garcia
 * @version      $Id $
 */

$modversion['name']        = _MI_WB_MD_NAME;
$modversion['description'] = _MI_WB_MD_DESC;
$modversion['author']      = "hsalazar";
$modversion['credits']     = "Catzwolf";
//$modversion['help'] = "help/wordbook.html";
$modversion['official'] = 0;
$modversion['image']    = "assets/images/wb_slogo.png";
$moduleDirName          = basename(__DIR__);
$modversion['dirname']  = $moduleDirName;
// v.1.17
$modversion['help']        = 'page=help';
$modversion['license']     = 'GNU GPL 2.0 or later';
$modversion['license_url'] = 'www.gnu.org/licenses/gpl-2.0.html';
$modversion["release"]     = "2006-11-30";

$modversion['dirmoduleadmin'] = 'Frameworks/moduleclasses/moduleadmin';
$modversion['sysicons16']     = 'Frameworks/moduleclasses/icons/16';
$modversion['sysicons32']     = 'Frameworks/moduleclasses/icons/32';
$modversion['modicons16']     = 'assets/images/icons/16';
$modversion['modicons32']     = 'assets/images/icons/32';

//about
$modversion['version']             = "2.00";
$modversion['module_status']       = 'Beta 1';
$modversion['release_date']        = '2016/01/02';
$modversion['module_website_url']  = 'www.xoops.org';
$modversion['module_website_name'] = 'XOOPS';
$modversion['min_php']             = '5.5';
$modversion['min_xoops']           = '2.5.7.2';
$modversion['min_admin']           = '1.1';
$modversion['min_db']              = array(
    'mysql'  => '5.0.7',
    'mysqli' => '5.0.7');

//$modversion["author_website_url"] = "http://dev.xoops.org/modules/xfmod/project/?wordbook";
//$modversion["author_website_name"] = "Wordbook on the Developers Forge";
$modversion["author_word"]   = ":-)";
$modversion["module_status"] = "Release Candidate 1";

$modversion["module_team"] = "hsalazar, mondarse, alama";
//$modversion["module_website_url"] = "http://dev.xoops.org/modules/xfmod/project/?wordbook";
//$modversion["module_website_name"] = "Wordbook on the Developers Forge";

$modversion['demo_site_url']  = "http://www.somesite.net/modules/wordbook/";
$modversion['demo_site_name'] = "";
$modversion['submit_bug']     = "http://xoops.org/modules/newbb";
$modversion['submit_feature'] = "https://github.com/XoopsModulesArchive/wordbook/issues";

// Admin things
$modversion['system_menu'] = 1;
$modversion['hasAdmin']    = 1;
$modversion['adminindex']  = "admin/index.php";
$modversion['adminmenu']   = "admin/menu.php";

// Sql file (must contain sql generated by phpMyAdmin or phpPgAdmin)
// All tables should not have any prefix!
$modversion['sqlfile']['mysql'] = "sql/mysql.sql";

// Tables created by sql file (without prefix!)
$modversion['tables'][0] = "wbcategories";
$modversion['tables'][1] = "wbentries";

// Search
$modversion['hasSearch']      = 1;
$modversion['search']['file'] = "include/search.inc.php";
$modversion['search']['func'] = "wb_search";

// Menu
global $xoopsUser, $xoopsDB, $xoopsModule;

$modversion['hasMain'] = 1;
$hModConfig            =& xoops_gethandler('config');
$hModule               =& xoops_gethandler('module');
if ($wordbookModule =& $hModule->getByDirname('wordbook')) {
    $wordbookConfig =& $hModConfig->getConfigsByCat(0, $wordbookModule->getVar('mid'));
    //if ( isset($wordbookConfig['catsinmenu']) && $wordbookConfig['anonpost'] == 1 )
    if (($xoopsUser && ($wordbookConfig['allowsubmit'] == 1)) || ($wordbookConfig['anonpost'] == 1)) {
        $modversion['sub'][1]['name'] = _MI_WB_SUB_SMNAME1;
        $modversion['sub'][1]['url']  = "submit.php";
    }
}
//if ($xoopsUser)
//  {
//  $modversion['sub'][1]['name'] = _MI_WB_SUB_SMNAME1;
//  $modversion['sub'][1]['url'] = "submit.php";
//  }
$modversion['sub'][2]['name'] = _MI_WB_SUB_SMNAME2;
$modversion['sub'][2]['url']  = "request.php";
$modversion['sub'][3]['name'] = _MI_WB_SUB_SMNAME3;
$modversion['sub'][3]['url']  = "search.php";

$sql        = $xoopsDB->query("SELECT categoryID, name FROM " . $xoopsDB->prefix("wbcategories") . " ");
$i          = 3;
$hModConfig =& xoops_gethandler('config');
$hModule    =& xoops_gethandler('module');
if ($wordbookModule =& $hModule->getByDirname('wordbook')) {
    $wordbookConfig =& $hModConfig->getConfigsByCat(0, $wordbookModule->getVar('mid'));
    if (isset($wordbookConfig['catsinmenu']) && $wordbookConfig['catsinmenu'] == 1) {
        while (list($categoryID, $name) = $xoopsDB->fetchRow($sql)) {
            $modversion['sub'][$i]['name'] = $name;
            $modversion['sub'][$i]['url']  = "category.php?categoryID=" . $categoryID . "";
            $i++;
        }
    }
}
// blocks
$modversion['blocks'][1]['file']        = "entries_new.php";
$modversion['blocks'][1]['name']        = _MI_WB_ENTRIESNEW;
$modversion['blocks'][1]['description'] = "Shows new entries";
$modversion['blocks'][1]['show_func']   = "b_entries_new_show";
$modversion['blocks'][1]['edit_func']   = "b_entries_new_edit";
$modversion['blocks'][1]['options']     = "datesub|5";
$modversion['blocks'][1]['template']    = "entries_new.tpl";

$modversion['blocks'][2]['file']        = "entries_top.php";
$modversion['blocks'][2]['name']        = _MI_WB_ENTRIESTOP;
$modversion['blocks'][2]['description'] = "Shows popular entries";
$modversion['blocks'][2]['show_func']   = "b_entries_top_show";
$modversion['blocks'][2]['edit_func']   = "b_entries_top_edit";
$modversion['blocks'][2]['options']     = "counter|5";
$modversion['blocks'][2]['template']    = "entries_top.tpl";

$modversion['blocks'][3]['file']        = "random_term.php";
$modversion['blocks'][3]['name']        = _MI_WB_RANDOMTERM;
$modversion['blocks'][3]['description'] = "Shows a random term";
$modversion['blocks'][3]['show_func']   = "b_entries_random_show";
$modversion['blocks'][3]['template']    = "entries_random.tpl";

// Templates
$modversion['templates'][1]['file']        = 'wb_category.tpl';
$modversion['templates'][1]['description'] = 'Display categories';
$modversion['templates'][2]['file']        = 'wb_index.tpl';
$modversion['templates'][2]['description'] = 'Display index';
$modversion['templates'][3]['file']        = 'wb_entry.tpl';
$modversion['templates'][3]['description'] = 'Display entry';
$modversion['templates'][4]['file']        = 'wb_letter.tpl';
$modversion['templates'][4]['description'] = 'Display letter';
$modversion['templates'][5]['file']        = 'wb_search.tpl';
$modversion['templates'][5]['description'] = 'Search in glossary';
$modversion['templates'][6]['file']        = 'wb_request.tpl';
$modversion['templates'][6]['description'] = 'Request a definition';
$modversion['templates'][7]['file']        = 'wb_submit.tpl';
$modversion['templates'][7]['description'] = 'Submit a definition';
// 1.17
$modversion['templates'][8]['file']        = "wb_adminmenu.tpl";
$modversion['templates'][8]['description'] = '(Admin) Tabs bar for administration pages';

// Config Settings (only for modules that need config settings generated automatically)
$cf                        = 1;
$modversion['config'][$cf] = array(
    'name'        => 'allowsubmit',
    'title'       => '_MI_WB_ALLOWSUBMIT',
    'description' => '_MI_WB_ALLOWSUBMITDSC',
    'formtype'    => 'yesno',
    'valuetype'   => 'int',
    'default'     => 1);

$cf++;
$modversion['config'][$cf] = array(
    'name'        => 'anonpost',
    'title'       => '_MI_WB_ANONSUBMIT',
    'description' => '_MI_WB_ANONSUBMITDSC',
    'formtype'    => 'yesno',
    'valuetype'   => 'int',
    'default'     => 0);

$cf++;
$modversion['config'][$cf] = array(
    'name'        => 'dateformat',
    'title'       => '_MI_WB_DATEFORMAT',
    'description' => '_MI_WB_DATEFORMATDSC',
    'formtype'    => 'textbox',
    'valuetype'   => 'text',
    'default'     => 'd-M-Y H:i');

$cf++;
$modversion['config'][$cf] = array(
    'name'        => 'perpage',
    'title'       => '_MI_WB_PERPAGE',
    'description' => '_MI_WB_PERPAGEDSC',
    'formtype'    => 'select',
    'valuetype'   => 'int',
    'default'     => 10,
    'options'     => array('5' => 5, '10' => 10, '15' => 15, '20' => 20, '25' => 25, '30' => 30, '50' => 50));

$cf++;
$modversion['config'][$cf] = array(
    'name'        => 'indexperpage',
    'title'       => '_MI_WB_PERPAGEINDEX',
    'description' => '_MI_WB_PERPAGEINDEXDSC',
    'formtype'    => 'select',
    'valuetype'   => 'int',
    'default'     => 10,
    'options'     => array('5' => 5, '10' => 10, '15' => 15, '20' => 20, '25' => 25, '30' => 30, '50' => 50));

$cf++;
$modversion['config'][$cf] = array(
    'name'        => 'autoapprove',
    'title'       => '_MI_WB_AUTOAPPROVE',
    'description' => '_MI_WB_AUTOAPPROVEDSC',
    'formtype'    => 'yesno',
    'valuetype'   => 'int',
    'default'     => 0);

$cf++;
$modversion['config'][$cf] = array(
    'name'        => 'multicats',
    'title'       => '_MI_WB_MULTICATS',
    'description' => '_MI_WB_MULTICATSDSC',
    'formtype'    => 'yesno',
    'valuetype'   => 'int',
    'default'     => 1);

$cf++;
$modversion['config'][$cf] = array(
    'name'        => 'catsinmenu',
    'title'       => '_MI_WB_CATSINMENU',
    'description' => '_MI_WB_CATSINMENUDSC',
    'formtype'    => 'yesno',
    'valuetype'   => 'int',
    'default'     => 0);

$cf++;
$modversion['config'][$cf] = array(
    'name'        => 'catsperindex',
    'title'       => '_MI_WB_CATSPERINDEX',
    'description' => '_MI_WB_CATSPERINDEXDSC',
    'formtype'    => 'select',
    'valuetype'   => 'int',
    'default'     => 3,
    'options'     => array('1' => 1, '3' => 3, '5' => 5, '10' => 10, '15' => 15, '20' => 20, '25' => 25, '30' => 30, '50' => 50));

$cf++;
$modversion['config'][$cf] = array(
    'name'        => 'adminhits',
    'title'       => '_MI_WB_ALLOWADMINHITS',
    'description' => '_MI_WB_ALLOWADMINHITSDSC',
    'formtype'    => 'yesno',
    'valuetype'   => 'int',
    'default'     => 0);

$cf++;
$modversion['config'][$cf] = array(
    'name'        => 'mailtoadmin',
    'title'       => '_MI_WB_MAILTOADMIN',
    'description' => '_MI_WB_MAILTOADMINDSC',
    'formtype'    => 'yesno',
    'valuetype'   => 'int',
    'default'     => 1);

$cf++;
$modversion['config'][$cf] = array(
    'name'        => 'rndlength',
    'title'       => '_MI_WB_RANDOMLENGTH',
    'description' => '_MI_WB_RANDOMLENGTHDSC',
    'formtype'    => 'textbox',
    'valuetype'   => 'int',
    'default'     => 150);

$cf++; //13
$modversion['config'][$cf] = array(
    'name'        => 'linkterms',
    'title'       => '_MI_WB_LINKTERMS',
    'description' => '_MI_WB_LINKTERMSDSC',
    'formtype'    => 'yesno',
    'valuetype'   => 'int',
    'default'     => 1);

//Comments (Mondarse)
$modversion['hasComments']          = 1;
$modversion['comments']['itemName'] = 'entryID';
$modversion['comments']['pageName'] = 'entry.php';

$modversion['comments']['callbackFile']        = 'include/comment_functions.php';
$modversion['comments']['callback']['approve'] = 'wordbook_com_approve';
$modversion['comments']['callback']['update']  = 'wordbook_com_update';
//Comments (Mondarse)

// On Update
if (!empty($_POST['fct']) && !empty($_POST['op']) && $_POST['fct'] === 'modulesadmin' && $_POST['op'] === 'update_ok' && $_POST['dirname'] == $modversion['dirname']) {
    include __DIR__ . "/include/onupdate.inc.php";
}
